### ISSUE TITLE
Making this accessible to screen readers, assistive tech. Umbrella issue.

### ISSUE THREAD

**Issue by flarnie-chegg** [catcarbonell/popop#1]
Hi @catcarbonell - 
Saw your tweet about this and I like your modal! With some small adjustments, we can enable people who use a screen reader, keyboard, and other assistive technology to better interact with this modal.

Here are the steps:
* When the modal opens, we need to move the focus to some element inside the modal. If we don't, then a keyboard user is stuck in the background page and may not be able to get into the modal to activate buttons. Screen reader users likewise will not be alerted to the modal opening and will be stuck in the background. (explained in https://github.com/catcarbonell/popop/issues/2)
* When the modal closes, we need to move the focus back where it was before the modal opened, if possible. Otherwise if the focus is inside the modal, and then the modal is hidden or deleted from the DOM, the focus can be "lost" and keyboard users will be stuck or have to start at the top of the page again. (explained in https://github.com/catcarbonell/popop/issues/2)
* While the modal is open, the "background" page should be blocked from all interactions. This prevents a non-sighted user from getting "stuck" in the "background" page when you intend them to interact with the modal. (explained in https://github.com/catcarbonell/popop/issues/3)
*  A modal should have the `role="dialog"` with an `aria-labelled-by` attribute; we could check with JS that has been done, or accept arguments for the 'label' id and then set those properties ourselves when the modal first opens.
* **Edit:** Nearly forgot - the modal should close when the user presses 'esc'.

More complete guidelines are here - https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

I'd like to open a separate issue for each of these, and can give you a "cheat sheet" for solving each one there.

**Comment by flarnie-chegg** [comment]
Out of time for today but this is a good starting point. May add more later.
